torch/csrc/distributed/c10d/sequence_num.cpp

```diff
@@ -10,52 +10,61 @@ SequenceNum::SequenceNum() : num_(c10::nullopt) {}
 SequenceNum::SequenceNum(const uint64_t num) : num_(num) {}
 
 SequenceNum::SequenceNum(const SequenceNum& other) {
-  if (!other.isSet()) {
-    num_ = c10::nullopt;
-  } else {
-    num_ = other.get();
-  }
+  num_.withLock([&](auto& num) {
+    if (!other.isSet()) {
+      num = c10::nullopt;
+    } else {
+      num = other.get();
+    }
+  });
```

Comment:
This is okay to do under a lock (not necessary), but there will be no contention since this the constructor. The new API mandates that this be done under a lock, so I wanted to highlight it.